src/Button/Button.svelte

```diff
@@ -77,8 +69,14 @@
   /** Obtain a reference to the HTML element */
   export let ref = null;
 
+  /**
+   * Button, anchor, or div attributes
+   * @type {import('svelte/elements').HTMLAnchorAttributes |
+   * import('svelte/elements').HTMLButtonAttributes | import('svelte/elements').HTMLAttributes}
+   */
+  export let buttonAttributes = {};
```

Comment:
I am not a fan of this, at least not in this form/for this component.
- It's unwieldy, you repeat "button" which is already part of the element, maybe it could just be `attributes` if it refers to the root element
- `Button` primarily is just a wrapper around a single element, I would prefer to pass any unknown properties automatically via `$$restProps`
- If an attributes property is used, there should at least also be `iconAttributes`.
- It is inconsistent, some attributes, like `disabled`, have their own property.

Reply:
These are all good points. I think it's fair to revise #1621 accordingly.

> maybe it could just be `attributes` if it refers to the root element

I think `attributes` works. The main benefit of a dedicated prop over `$$restProps` is for more ergonomic autocomplete. Only component props would appear in the IDE, and element attributes would be contained within a dedicated prop such as `attributes`.

> I would prefer to pass any unknown properties automatically via `$$restProps`

Given the autocomplete trade-off above, I'm leaning towards `attributes`, but I'm also torn because this:

```html
<Button kind="secondary" type="submit" data-test-id="abc123">Save</button>
```

Looks nicer than this:

```html
<Button kind="secondary" type="submit" attributes={{ 'data-test-id': "abc123" }}>
  Save
</button>
```

> there should at least also be `iconAttributes`.

Agreed.

We have this note in our standardization ticket:

>HTML attributes are provided via Type declarations and should not be included as explicit props unless it has special treatment as part of a component.
>
>Appropriate: `disabled` for `<Button>`
>Inappropriate: `aria-label` for `<TextInput>`

I suppose `disabled` has special treatment depending on usage of `as` or `href`. For that reason, I think it deserves to be a prop.